graf2d/gpad/src/TCanvas.cxx

```diff
@@ -779,7 +779,7 @@ void TCanvas::Close(Option_t *option)
 {
    TPad    *padsave = (TPad*)gPad;
    TCanvas *cansave = 0;
-   if (padsave) cansave = (TCanvas*)gPad->GetCanvas();
+   if (padsave && gPad->TestBit(kNotDeleted)) cansave = (TCanvas*)gPad->GetCanvas();
```

Comment:
This is sub-optiomal has this test (TestBit(kNoDeleted)) can be foiled if the memory re-used between the deletion and the execution.  So question "why" in this case is gPad still pointing to a deleted object?

Reply:
the pad "c" is deleted explicitly by the macro pad4.C and deleted again when the canvas C is closed. gPad was set to "c" and was not set to null when c  was deleted (I guess). So a better change might be to set gPad to null when a pad equal to gPad is deleted. Let me see if a such mechanism can be done.

Reply:
``if (this==gPad) gPad=nullptr;``  in the ``~TPad`` ?
or somathing like 

    ``if (this==gPad) gPad=fMother;``  in the also in  ``~TPad``

Reply:
I do not know. testing now.

Reply:
If I undo the change in `TCanvas::Close()`  and change the `~TPad()` the following way:
```
TPad::~TPad()
{
   if (!TestBit(kNotDeleted)) return;
   if (gPad == this) gPad = nullptr;  // <<<< new test

```

Then it also fixes the problem. ie: pad4.cxx works and stressGraphics also.

Reply:
Wouldnt fMother make more sense?

Reply:
It sounds more logical to test `gPad` seems to me. The ctor delete `this` pad and if `this` is `gPad`, then `gPad` must be null also.  I am not sure to understand the `fMother` logic.

Reply:
I think that is just a convention. 
One can say the logic is that after the deletion of a subpad, gPad points to the parent pad. But null is another option.

Reply:
I guess we can leave it like that for the time being.